### ISSUE TITLE
Crash while computing coverage info

### ISSUE THREAD

**Issue by BradSwain** [coderrect-inc/OpenRace#317]
**Describe the bug**
I encountered a segfault while running the tool on a dataracebench case. The stack dump looks like it crashed during `Coverage::summarize()`.

**Expected behavior**
No crash

**To Reproduce**
I built openrace in release mode and run it on DRB108
```shell
$ openrace tests/data/integration/dataracebench/DRB108-atomic-orig-no.ll
```

**Additional context**
Here is the stack dump

```shell
Stack dump:
0.      Program arguments: openrace /OpenRace/tests/data/integration/dataracebench/DRB108-atomic-orig-no.ll 
 #0 0x000055f84c8b1ace llvm::sys::PrintStackTrace(llvm::raw_ostream&) (/home/brad/coderrect-inc/OpenRace/build/bin/openrace+0x698ace)
 #1 0x000055f84c8af814 llvm::sys::RunSignalHandlers() (OpenRace/build/bin/openrace+0x696814)
 #2 0x000055f84c8af958 SignalHandler(int) (OpenRace/build/bin/openrace+0x696958)
 #3 0x00007f355d00a3c0 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x153c0)
 #4 0x000055f84c4d52e8 race::Coverage::summarize() (.part.0) (OpenRace/build/bin/openrace+0x2bc2e8)
 #5 0x000055f84c4d54eb race::Coverage::Coverage(race::ProgramTrace const&) (OpenRace/build/bin/openrace+0x2bc4eb)
 #6 0x000055f84c491552 race::detectRaces(llvm::Module*, race::DetectRaceConfig) (OpenRace/build/bin/openrace+0x278552)
 #7 0x000055f84c44d3e4 main (OpenRace/build/bin/openrace+0x2343e4)
 #8 0x00007f355cadc0b3 __libc_start_main /build/glibc-eX1tMB/glibc-2.31/csu/../csu/libc-start.c:342:3
 #9 0x000055f84c48262e _start (OpenRace/build/bin/openrace+0x26962e)
Segmentation fault (core dumped)
```

**Comment by bz-liu** [comment]
Fixed. This bug is because when there is only atomic read/write in a omp fork thread, the thread will have no events (i.e., empty trace). 

Is there a way to automatically conduct the coverage computation for all tests in openrace?

**Comment by BradSwain** [comment]
> Is there a way to automatically conduct the coverage computation for all tests in openrace?

Do you just want to test that there is no crash, or see the coverage info?

If just checking for no crash, we can run the tests with coverage enabled:

https://github.com/coderrect-inc/OpenRace/blob/3c820af62bfc01721d1aade297100c3c5aff1674/tests/helpers/ReportChecking.cpp#L137

**Comment by bz-liu** [comment]
I would like to test no no crash, and I though this change of `.doCoverage = true` can print out the coverage info. Seems like no. THX.

**Comment by BradSwain** [comment]
Fixed in #319
